### ISSUE TITLE
'pip install ansible==' installs the wrong version in a virtualenv

### ISSUE THREAD

**Issue by vossenjp** [ansible/ansible#75141]
### Summary

When I create an *empty* Python3 virtual environment, then use `pip install ansible==3.4.0` and/or `pip install ansible==2.10.7` to install Ansible into it, I end up with `ansible 2.10.11`.  I have consistently had this problem for various installs since early 2021 using Linux Mint-19.3, though the end results wasn't always 2.10.11.  Over the weekend I upgraded to Mint-20.0 then 20.1 and the problem remains.

i have read https://groups.google.com/g/ansible-devel/c/aDtLYNnZ8LQ which says "Due to a limitation in pip, if you are upgrading from Ansible 2.9 (or earlier), you need to uninstall ansible before installing the 3.x version" but that's incredibly frustratingly useless because:
1. WHAT "limitation"?!?
2. That work-around is pointless and impossible in a brand new *empty virtualenv*

I found https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-ansible-core unhelpful for this issue.

I can't believe I'm the only person installing Ansible only into virtenvs, and having this problem, but I've failed to find anything else relevant.

I could use a PPA on Mint, but I don't want to because I must use virtualenv's on servers elsewhere and I want the flexibility to have multiple versions that *I* choose (which this bug is frustratingly precluding!) that can match various server versions.

### Issue Type

Bug Report

### Component Name

install

### Ansible Version

```console
The wrong one
```

### Configuration

```console
N/A
```

### OS / Environment

Previously Linux Mint-19.3
Now Linux Mint 20.1
Python 3.8.5
`pip` is NOT installed outside the virtenvs, though `python-pip-whl-20.0.2-5ubuntu1.5` is.
Ansible is NOT installed outside the virtenvs either

### Steps to Reproduce

```
[jp@hostname:T15(15):L3:C5025:J0:2021-06-29_17:17:32_EDT]
/home/jp/venv$ rm -rf ansible-3.4.0 ~/.cache/pip

[jp@hostname:T15(15):L3:C5028:J0:2021-06-29_17:18:08_EDT]
/home/jp/venv$ python3 -m virtualenv -p python3 ~/venv/ansible-3.4.0
created virtual environment CPython3.8.5.final.0-64 in 242ms
  creator CPython3Posix(dest=/home/jp/venv/ansible-3.4.0, clear=False, global=False)
  seeder FromAppData(download=False, pip=latest, setuptools=latest, wheel=latest, pkg_resources=latest, via=copy, app_data_dir=/home/jp/.local/share/virtualenv/seed-app-data/v1.0.1.debian.1)
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

[jp@hostname:T15(15):L3:C5028:J0:2021-06-29_17:18:19_EDT]
/home/jp/venv$ perl -i -pe 's/PS1="\(`basename \\"\$VIRTUAL_ENV\\"`\) ?\$\{PS1-\}"/PS1="\\n(`basename \"\$VIRTUAL_ENV\"`)\$PS1"/;' ~/venv/ansible-3.4.0/bin/activate

[jp@hostname:T15(15):L3:C5028:J0:2021-06-29_17:18:23_EDT]
/home/jp/venv$ source ~/venv/ansible-3.4.0/bin/activate

(ansible-3.4.0)
[jp@hostname:T15(15):L3:C5028:J0:2021-06-29_17:18:29_EDT]
/home/jp/venv$ pip install ansible==3.4.0
Collecting ansible==3.4.0
  Downloading ansible-3.4.0.tar.gz (31.9 MB)
     |████████████████████████████████| 31.9 MB 24 kB/s
Collecting ansible-base<2.11,>=2.10.9
  Downloading ansible-base-2.10.11.tar.gz (6.0 MB)
     |████████████████████████████████| 6.0 MB 9.0 MB/s
Collecting PyYAML
  Downloading PyYAML-5.4.1-cp38-cp38-manylinux1_x86_64.whl (662 kB)
     |████████████████████████████████| 662 kB 12.1 MB/s
Collecting cryptography
  Downloading cryptography-3.4.7-cp36-abi3-manylinux2014_x86_64.whl (3.2 MB)
     |████████████████████████████████| 3.2 MB 12.0 MB/s
Collecting jinja2
  Downloading Jinja2-3.0.1-py3-none-any.whl (133 kB)
     |████████████████████████████████| 133 kB 11.6 MB/s
Collecting packaging
  Downloading packaging-20.9-py2.py3-none-any.whl (40 kB)
     |████████████████████████████████| 40 kB 4.8 MB/s
Collecting cffi>=1.12
  Downloading cffi-1.14.5-cp38-cp38-manylinux1_x86_64.whl (411 kB)
     |████████████████████████████████| 411 kB 12.8 MB/s
Collecting MarkupSafe>=2.0
  Downloading MarkupSafe-2.0.1-cp38-cp38-manylinux2010_x86_64.whl (30 kB)
Collecting pyparsing>=2.0.2
  Downloading pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
     |████████████████████████████████| 67 kB 2.4 MB/s
Collecting pycparser
  Downloading pycparser-2.20-py2.py3-none-any.whl (112 kB)
     |████████████████████████████████| 112 kB 9.8 MB/s
Building wheels for collected packages: ansible, ansible-base
  Building wheel for ansible (setup.py) ... done
  Created wheel for ansible: filename=ansible-3.4.0-py3-none-any.whl size=52834537 sha256=942c83b5da706b4ea9fdc900c5be21d0b1ffd7428e29b3fe68c9f10653452414
  Stored in directory: /home/jp/.cache/pip/wheels/8a/73/16/ad7b1b2705d83ec00ec567486db31547c07398807179b939de
  Building wheel for ansible-base (setup.py) ... done
  Created wheel for ansible-base: filename=ansible_base-2.10.11-py3-none-any.whl size=1877312 sha256=6e42e22760398a28766996e2ce92d65809804e93935512b6cd7d8da689d9f62b
  Stored in directory: /home/jp/.cache/pip/wheels/ec/62/a4/8b5f0cef3c8739b85a41d3da3cc710c1fd6d8b2e615e3d7d49
Successfully built ansible ansible-base
Installing collected packages: PyYAML, pycparser, cffi, cryptography, MarkupSafe, jinja2, pyparsing, packaging, ansible-base, ansible
Successfully installed MarkupSafe-2.0.1 PyYAML-5.4.1 ansible-3.4.0 ansible-base-2.10.11 cffi-1.14.5 cryptography-3.4.7 jinja2-3.0.1 packaging-20.9 pycparser-2.20 pyparsing-2.4.7

(ansible-3.4.0)
[jp@hostname:T15(15):L3:C5028:J0:2021-06-29_17:21:30_EDT]
/home/jp/venv$ ansible --version
ansible 2.10.11
  config file = None
  configured module search path = ['/home/jp/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jp/venv/ansible-3.4.0/lib/python3.8/site-packages/ansible
  executable location = /home/jp/venv/ansible-3.4.0/bin/ansible
  python version = 3.8.5 (default, May 27 2021, 13:30:53) [GCC 9.3.0]
```

### Expected Results

I expect `pip install ansible==3.4.0` to install ansible-3.4.0 and `pip install ansible==2.10.7` to install 2.10.7.

### Actual Results

```console
$ ansible --version
ansible 2.10.11
```
That's actually quite infuriating, since what I'd done previously, and thought it had worked, was `pip install ansible==2.10.7`.  But THAT actually installed ansible 2.10.11 and I didn't catch that THAT one was wrong too until just now.
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* `lib/ansible/modules/packaging/os/installp.py`

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/modules/packaging/os/installp.py -> https://github.com/ansible-collections/community.general (https://galaxy.ansible.com/community/general)

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md

**Comment by jborean93** [comment]
The output of `ansible --version` reflects the version of `ansible-base|core` which is the engine component of Ansible. The `ansible` PyPI package includes both `ansible-base|core` as well as a whole bunch of collections to provide a consistent experience from 2.9 and newer. This is covered in more detail at https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#selecting-an-ansible-artifact-and-version-to-install.

**Comment by vossenjp** [comment]
Now I'm even more confused:
```
$ pip install ansible==2.10.11
ERROR: Could not find a version that satisfies the requirement ansible==2.10.11 (from versions: 1.0, 1.1, 1.2, 1.2.1, 1.2.2, 1.2.3, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.4, 1.4.1, 1.4.2, 1.4.3, 1.4.4, 1.4.5, 1.5, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.5.5, 1.6, 1.6.1, 1.6.2, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.6.7, 1.6.8, 1.6.9, 1.6.10, 1.7, 1.7.1, 1.7.2, 1.8, 1.8.1, 1.8.2, 1.8.3, 1.8.4, 1.9.0.1, 1.9.1, 1.9.2, 1.9.3, 1.9.4, 1.9.5, 1.9.6, 2.0.0.0, 2.0.0.1, 2.0.0.2, 2.0.1.0, 2.0.2.0, 2.1.0.0, 2.1.1.0, 2.1.2.0, 2.1.3.0, 2.1.4.0, 2.1.5.0, 2.1.6.0, 2.2.0.0, 2.2.1.0, 2.2.2.0, 2.2.3.0, 2.3.0.0, 2.3.1.0, 2.3.2.0, 2.3.3.0, 2.4.0.0, 2.4.1.0, 2.4.2.0, 2.4.3.0, 2.4.4.0, 2.4.5.0, 2.4.6.0, 2.5.0a1, 2.5.0b1, 2.5.0b2, 2.5.0rc1, 2.5.0rc2, 2.5.0rc3, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.5.4, 2.5.5, 2.5.6, 2.5.7, 2.5.8, 2.5.9, 2.5.10, 2.5.11, 2.5.12, 2.5.13, 2.5.14, 2.5.15, 2.6.0a1, 2.6.0a2, 2.6.0rc1, 2.6.0rc2, 2.6.0rc3, 2.6.0rc4, 2.6.0rc5, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.6.6, 2.6.7, 2.6.8, 2.6.9, 2.6.10, 2.6.11, 2.6.12, 2.6.13, 2.6.14, 2.6.15, 2.6.16, 2.6.17, 2.6.18, 2.6.19, 2.6.20, 2.7.0.dev0, 2.7.0a1, 2.7.0b1, 2.7.0rc1, 2.7.0rc2, 2.7.0rc3, 2.7.0rc4, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.7.5, 2.7.6, 2.7.7, 2.7.8, 2.7.9, 2.7.10, 2.7.11, 2.7.12, 2.7.13, 2.7.14, 2.7.15, 2.7.16, 2.7.17, 2.7.18, 2.8.0a1, 2.8.0b1, 2.8.0rc1, 2.8.0rc2, 2.8.0rc3, 2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.8.4, 2.8.5, 2.8.6, 2.8.7, 2.8.8, 2.8.9, 2.8.10, 2.8.11, 2.8.12, 2.8.13, 2.8.14, 2.8.15, 2.8.16rc1, 2.8.16, 2.8.17rc1, 2.8.17, 2.8.18rc1, 2.8.18, 2.8.19rc1, 2.8.19, 2.8.20rc1, 2.8.20, 2.9.0b1, 2.9.0rc1, 2.9.0rc2, 2.9.0rc3, 2.9.0rc4, 2.9.0rc5, 2.9.0, 2.9.1, 2.9.2, 2.9.3, 2.9.4, 2.9.5, 2.9.6, 2.9.7, 2.9.8, 2.9.9, 2.9.10, 2.9.11, 2.9.12, 2.9.13, 2.9.14rc1, 2.9.14, 2.9.15rc1, 2.9.15, 2.9.16rc1, 2.9.16, 2.9.17rc1, 2.9.17, 2.9.18rc1, 2.9.18, 2.9.19rc1, 2.9.19, 2.9.20rc1, 2.9.20, 2.9.21rc1, 2.9.21, 2.9.22rc1, 2.9.22, 2.9.23rc1, 2.9.23, 2.10.0a1, 2.10.0a2, 2.10.0a3, 2.10.0a4, 2.10.0a5, 2.10.0a6, 2.10.0a7, 2.10.0a8, 2.10.0a9, 2.10.0b1, 2.10.0b2, 2.10.0rc1, 2.10.0, 2.10.1, 2.10.2, 2.10.3, 2.10.4, 2.10.5, 2.10.6, 2.10.7, 3.0.0b1, 3.0.0rc1, 3.0.0, 3.1.0, 3.2.0, 3.3.0, 3.4.0, 4.0.0a1, 4.0.0a2, 4.0.0a3, 4.0.0a4, 4.0.0b1, 4.0.0b2, 4.0.0rc1, 4.0.0, 4.1.0)
ERROR: No matching distribution found for ansible==2.10.11
```

It looks like I'm missing something really fundamental here, but I have no idea what it is...  And OK, thanks @jborean93 I guess I'm just really behind the times.  But then if `ansible --version` lies, how do I know what is *really* running?  I've got to say that all hurts my head.

**Comment by vossenjp** [comment]
I also have to say that I spent some time trying to make this report clear and useful, and I find it rude and quite disappointing that it was summarily closed--not moved, but closed--by a bot.  If we shouldn't create bug reports here...don't let us create bug reports here!

**Comment by jborean93** [comment]
This thing is this isn't a bug report, we do have docs that state this information and I've tried to explain it a bit more after the bot was closed. Just to repeat the `ansible` package on PyPI does not reflect the version of the Ansible engine which is what `ansible --version` reports. It's a combination of the engine itself + multiple other packages, those being various Ansible collections.

Before 2.9 things were all the same, since 2.10 it's been split apart.

TLDR:

* If you want the package and how it was in 2.9 and ealier then do the same thing before and install the `ansible` package from PyPI
  * `ansible-core` is a dep of this package and is picked up automatically
  * Typically each major version of the `ansible` package targets a specific engine/core version, e.g. `ansible 3.x.y` has `ansible-core 2.10`, `4.x.y` has `2.11`, etc
* If you wish to target a specific engine version and manage the collections yourself use the `ansible-core` package from PyPI
  * `ansible-galaxy collection install` can install specific collections
* and for things to act the same as before use the `ansible` package from PyPI. Typically each major version corresponds to a 2.x engine release, . If you want a specific engine version then use `ansible-core` from PyPI and `ansible-galaxy collection install` to install your collections.
